### ISSUE TITLE
Kattis

### ISSUE THREAD

**Issue by mzuenni** [mzuenni/kattis-export#1]
I added all missing gcpc's and nwerc that i have @niemela what needs to be done to get them in kattis?

**Comment by mzuenni** [comment]
@mpsijm did you already prepare a version of last years nwerc problems for kattis upload?

**Comment by mpsijm** [comment]
@mzuenni Yes I did, at Ragnar's GitLab server, in February. They provided feedback and I fixed it, but haven't heard from them since (I also sent a reminder email in March) :upside_down_face:

**Comment by niemela** [comment]
> I added all missing gcpc's and nwerc that i have @niemela what needs to be done to get them in kattis?

I've started uploading them, so far I have run into these minor changes needed:

- UUIDs needs to be added. I will create and push them to the repo, I would suggest that you send them upstream to your copy as well.
- The short name needs to be changed when it collides with an existing Kattis problem. I will do that, and since this is a specific *kattis* export repo, I will push it. That said, there is no real reason why you need to send that upstream.
- keywords must be *space* separated, not *comma* separated. In theory you might want to have a comma *in* a keyword (and that's technically allowed), but in practice it's always an error, so we simply disallow commas. This will be fixed in the next version of the spec, keywords will be a proper YAML list.

**Comment by mzuenni** [comment]
@mpsijm I only pushed the most recent version from the tudelf git, maybe you want to add your version here so we don't have to make changes twice?

**Comment by mpsijm** [comment]
@mzuenni I've added the changes that I had to make after the upload test in February :slightly_smiling_face: 

Can I also ask the Kattis people to look at the repository on Ragnar's server again, so we can also get the BAPC problems on Kattis? @ghamerly opened an issue with comments there at the time, but @niemela should also have access.

**Comment by mzuenni** [comment]
@niemela any further updates here?

**Comment by mzuenni** [comment]
@niemela on my machine problemtools is fine with gcpc 2021, 2022 and 2025 now. (Only complaints about sensitive timelimit but these can be ignored for the upload to kattis). Let me know if there are more issues.

**Comment by niemela** [comment]
2021 ans 2022 are installed: https://open.kattis.com/problem-sources/German%20Collegiate%20Programming%20Contest%20%28GCPC%29%202021, https://open.kattis.com/problem-sources/German%20Collegiate%20Programming%20Contest%202022

For 2025, the problem jumbledpackets uses the new version of the problem format, which is not supported in production yet :(.

**Comment by mzuenni** [comment]
yeah that one is a multipass problem... so maybe just skip it for now? (i do not know how to write a multipass problem for kattis ^^')

**Comment by mzuenni** [comment]
@niemela 2021 only has 5 problems?

**Comment by paul-wild** [comment]
Thanks for setting this up!

Problem I (Island Urbanism) from GCPC 2025 is also missing

**Comment by mzuenni** [comment]
2021 looks fine now, thanks :)
So only GCPC 2025 Island Urbanism is missing (that one is not the multipass problem).
Additionally, we have the kind request to change the two contest names from 
`German Collegiate Programming Contest 2022`, `German Collegiate Programming Contest 2025` to `German Collegiate Programming Contest (GCPC) 2022`, `German Collegiate Programming Contest (GCPC) 2025`. So add the `(GCPC)` before the year.

@paul-wild will probably provide older gcpc's when he has time.

**Comment by mzuenni** [comment]
@niemela besides the points above, the git now also has gcpc2019 ready for Kattis.

2018 and 2017 still need some work
